azure/scope/managedcontrolplane_test.go

```diff
@@ -34,6 +34,66 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
+func TestManagedControlPlaneScope_OutboundType(t *testing.T) {
+	scheme := runtime.NewScheme()
+	_ = expv1.AddToScheme(scheme)
+	_ = infrav1.AddToScheme(scheme)
+	explicitOutboundType := infrav1.ManagedControlPlaneOutboundTypeUserDefinedRouting
+	cases := []struct {
+		Name     string
+		Input    ManagedControlPlaneScopeParams
+		Expected bool
+	}{
+		{
+			Name: "With Explicit OutboundType defined",
+			Input: ManagedControlPlaneScopeParams{
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+				},
+				ControlPlane: &infrav1.AzureManagedControlPlane{
+					Spec: infrav1.AzureManagedControlPlaneSpec{
+						SubscriptionID: "00000000-0000-0000-0000-000000000000",
+						OutboundType:   &explicitOutboundType,
+					},
+				},
+			},
+			Expected: false,
+		},
+		{
+			Name: "Without OutboundType defined",
+			Input: ManagedControlPlaneScopeParams{
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+				},
+				ControlPlane: &infrav1.AzureManagedControlPlane{
+					Spec: infrav1.AzureManagedControlPlaneSpec{
+						SubscriptionID: "00000000-0000-0000-0000-000000000000",
+					},
+				},
+			},
+			Expected: true,
+		},
+	}
+	for _, c := range cases {
+		c := c
+		t.Run(c.Name, func(t *testing.T) {
+			g := NewWithT(t)
+			fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(c.Input.ControlPlane).Build()
+			c.Input.Client = fakeClient
+			s, err := NewManagedControlPlaneScope(context.TODO(), c.Input)
+			g.Expect(err).To(Succeed())
+			result := s.ControlPlane.Spec.OutboundType == nil
```

Comment:
To ensure this is exercising the corresponding code change, let's have this test call `s.ManagedClusterSpec()` and get the value we're testing against from there. One of the tests below does something like that: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/84efdfbccbf5a57f7ed9847d3667abaea859dba9/azure/scope/managedcontrolplane_test.go#L326-L329